javascript/modules/client/src/client.ts

```diff
@@ -282,32 +277,93 @@ export class Client extends ClientCore implements IClient {
       override.value = amount;
     }
 
-    return daoInstance.estimateGas.deposit(
-      tokenAddress,
-      amount,
-      reference,
-      override,
-    ).then((gasLimit) => {
-      return this.web3.getApproximateGasFee(gasLimit.toBigInt());
+    return daoInstance.estimateGas
+      .deposit(tokenAddress, amount, reference, override)
+      .then(gasLimit => {
+        return this.web3.getApproximateGasFee(gasLimit.toBigInt());
+      });
+  }
+
+  //// PRIVATE METHODS METADATA
+
+  private async _getDaoMetadata(daoIdentifier: string): Promise<DaoMetadata> {
+    // TODO: Implement actual fetch logic using subgraph.
+    // Note: it would be nice if the client could be instantiated with dao identifier
+    // thereby removing the need to pass in the identifier for all the data retrieval methods
+
+    if (!daoIdentifier) {
+      throw new Error("A DAO identifier is needed");
+    }
+
+    // Generate DAO creation within the past year
+    const fromDate = new Date(
+      new Date().setFullYear(new Date().getFullYear() - 1)
+    ).getTime();
+
+    const dummyDaoNames = [
+      "Patito Dao",
+      "One World Dao",
+      "Sparta Dao",
+      "Yggdrasil Unite",
+    ];
+
+    return Promise.resolve({
+      ...(isAddress(daoIdentifier)
+        ? {
+            address: daoIdentifier,
+            name:
+              dummyDaoNames[
+                Math.floor(Math.random() * dummyDaoNames.length - 1)
+              ],
+          }
+        : {
+            address: "0x663ac3c648548eb8ccd292b41a8ff829631c846d",
+            name: daoIdentifier,
```

Comment:
We will not be indexing by DAO name. Indexing will be done by address only. If an ENS is provided, the SDK should resolve the ENS name and use that address. I don't understand the point on returning an input parameter itself

Reply:
Jordi, it's _dummy_ data. It's simply a way when the function is called with an address to return the data with the same address. Obviously this will change when implemented, no?